src/Control/Selective.hs

```diff
@@ -430,6 +432,14 @@ instance Selective ZipList where select = selectA
 -- a good example of a non-trivial selective functor which is not a monad.
 data Validation e a = Failure e | Success a deriving (Eq, Functor, Ord, Show)
 
+instance Foldable (Validation e) where
```

Comment:
Both `Foldable` and `Traversable` can be derived (using corresponding extensions), so let's do it.